### ISSUE TITLE
Allow Group Membership Changes through IdP Reconcile Lambda

### ISSUE THREAD

**Issue by robotdan** [FusionAuth/fusionauth-issues#1011]
### Description

The Reconcile Lambda simply modifies the user and then we make an update request. 

We can modify the Group Memberships in the Lambda invocation, but we ignore these changes. The User Update API does not modify Group Membership by design. 

In order to manage Group Membership properly in the Reconcile Lambda we will need to manage Group Memberships like we do Registrations in the reconcile workflow. 

Because we do allow group memberships during the User Create, technically the IdP Reconcile Lambda can add memberships if the user is not yet created. Subsequent requests will not work because they will be updates to the existing user.

**Comment by awoodsprim** [comment]
@robotdan Was looking for this exact issue.  I'd like the groups between my external provider and fusion auth to be synchronized.  Any idea if/when this would be considered for the product?

**Comment by mooreds** [comment]
@awoodsprim I don't think this is on the current near term roadmap, otherwise it would be assigned to a milestone.

You can see our general roadmap guidance here: https://fusionauth.io/docs/v1/tech/core-concepts/roadmap/ 

Please do vote this up with the 'thumbs up' emoji on the issue; that is the primary means we have of assessing community support for particular features.
